### PR TITLE
OpenCASCADE now works also in spacedim = 2

### PR DESCRIPTION
**Pull request by luca-heltai**

### CODE REVIEW

**Comment by luca-heltai** [comment]
This should be better. For some reasons git got confused by my two other commits, and mixed in David's commit in between...

**Comment by masterleinad** [comment]
There are some failing tests:
 - `opencascade/normal_projection_01.debug (Failed)`
 - `opencascade/create_tria_02.debug (Failed)`

**Comment by masterleinad** [comment]
This is the required patch:
```
diff --git a/tests/opencascade/create_tria_02.cc b/tests/opencascade/create_tria_02.cc
index b7165f3..9016fda 100644
--- a/tests/opencascade/create_tria_02.cc
+++ b/tests/opencascade/create_tria_02.cc
@@ -34,7 +34,7 @@ int main ()
   ps.push_back(Point<2>(0, 2));
   auto edge = interpolation_curve(ps);
 
-  BRepPrimAPI_MakeRevol::BRepPrimAPI_MakeRevol  revol(edge, gp_Ax1(gp_Pnt(0,0,0), gp_Dir(0,0,1)), -numbers::PI/2);
+  BRepPrimAPI_MakeRevol  revol(edge, gp_Ax1(gp_Pnt(0,0,0), gp_Dir(0,0,1)), -numbers::PI/2);
   revol.Build();
   auto sh = revol.Shape();
 
diff --git a/tests/opencascade/normal_projection_01.cc b/tests/opencascade/normal_projection_01.cc
index ddc02e7..2af56e7 100644
--- a/tests/opencascade/normal_projection_01.cc
+++ b/tests/opencascade/normal_projection_01.cc
@@ -42,7 +42,7 @@ int main ()
   std::ofstream logfile("output");
 
   gp_Pnt center(.5,.5,.5);
-  Standard_Real radius(Point<3>().distance(point(center)));
+  Standard_Real radius(Point<3>().distance(point<3>(center)));
 
 
   TopoDS_Face face = BRepPrimAPI_MakeSphere(center, radius);
```

File: source/opencascade/boundary_lib.cc [hunk 1]
```diff
@@ -248,9 +248,7 @@ namespace OpenCASCADE
     curve(curve_adaptor(sh)),
     tolerance(tolerance),
     length(shape_length(sh))
-  {
-    Assert(spacedim == 3, ExcNotImplemented());
-  }
+  {}
```
Comment (masterleinad) [inline source/opencascade/boundary_lib.cc]: Is `spacedim<2` allowed?

Comment (luca-heltai) [reply source/opencascade/boundary_lib.cc]: No, it is not instantiated for `spacedim < 2`. I shall put back an Assert with spacedim>=2

**Pull request merged**